::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Enea AB, Cavium and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 15 Mar 2016 11:33:52 +0100
Subject: [PATCH] Install vgabios (and link) for aarch64

vgabios is needed for standard VGA mode in AArch64 VMs, so
install it by default on AArch64 and create missing link
in </usr/share/qemu> to </usr/share/vgabios/vgabios.bin>.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Signed-off-by: Stanislaw Kardach <stanislaw.kardach@cavium.com>
---
 deployment/puppet/openstack_tasks/manifests/roles/compute.pp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/deployment/puppet/openstack_tasks/manifests/roles/compute.pp b/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
index 4240126..b18fa7b 100644
--- a/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
+++ b/deployment/puppet/openstack_tasks/manifests/roles/compute.pp
@@ -409,6 +409,14 @@ class openstack_tasks::roles::compute {
     libvirt_service_name                       => 'libvirt-bin',
     virtlock_service_name                      => 'virtlockd',
     virtlog_service_name                       => 'virtlogd',
+  } ->
+  package { 'vgabios':
+    ensure => present;
+  } ->
+  file { '/usr/share/qemu/vgabios-stdvga.bin':
+    ensure  => link,
+    target  => '/usr/share/vgabios/vgabios.bin',
+    replace => false,
   }

   class { '::nova::migration::libvirt':
